plugins/inputs/kube_inventory/statefulset.go

```diff
@@ -0,0 +1,46 @@
+package kube_inventory
+
+import (
+	"context"
+	"time"
+
+	"github.com/ericchiang/k8s/apis/apps/v1beta1"
+
+	"github.com/influxdata/telegraf"
+)
+
+func collectStatefulSets(ctx context.Context, acc telegraf.Accumulator, ki *KubernetesInventory) {
+	list, err := ki.client.getStatefulSets(ctx)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	for _, s := range list.Items {
+		if err = ki.gatherStatefulSet(*s, acc); err != nil {
+			acc.AddError(err)
+			return
+		}
+	}
+}
+
+func (ki *KubernetesInventory) gatherStatefulSet(s v1beta1.StatefulSet, acc telegraf.Accumulator) error {
+	status := s.Status
+	fields := map[string]interface{}{
+		"created":             time.Unix(s.Metadata.CreationTimestamp.GetSeconds(), int64(s.Metadata.CreationTimestamp.GetNanos())).UnixNano(),
+		"generation":          *s.Metadata.Generation,
```

Comment:
Suspicious pointer dereferencing...  Is this a double dereference?

Reply:
Not that I'm aware of, it's just dereferencing the `Generation`'s `*int64`.

<!-- We could convert these all to use the corresponding `Get` functions, but I don't like the pod_container's `GetTerminateReason` that results in an empty field. Which may/may not be the case on these. -->